pkgs/by-name/co/comfyui/package.nix

```diff
@@ -0,0 +1,164 @@
+{
+  lib,
+  python312,
+  linkFarm,
+  writers,
+  writeTextFile,
+  fetchFromGitHub,
+  stdenv,
+  symlinkJoin,
+  config,
+  modelsPath ? "/var/lib/comfyui/models",
+  inputPath ? "/var/lib/comfyui/input",
+  outputPath ? "/var/lib/comfyui/output",
+  tempPath ? "/var/lib/comfyui/temp",
+  userPath ? "/var/lib/comfyui/user",
+  customNodes ? [ ],
+  models ? {
+    checkpoints = { };
+    clip = { };
+    clip_vision = { };
+    configs = { };
+    controlnet = { };
+    embeddings = { };
+    upscale_modules = { };
+    vae = { };
+    vae_approx = { };
+  },
+}:
+
+let
+  config-data = {
+    comfyui = {
+      base_path = modelsPath;
+      checkpoints = "${modelsPath}/checkpoints";
+      clip = "${modelsPath}/clip";
+      clip_vision = "${modelsPath}/clip_vision";
+      configs = "${modelsPath}/configs";
+      controlnet = "${modelsPath}/controlnet";
+      diffusion_models = "${modelsPath}/diffusion_models";
+      embeddings = "${modelsPath}/embeddings";
+      loras = "${modelsPath}/loras";
+      style_models = "${modelsPath}/style_models";
+      text_encoders = "${modelsPath}/text_encoders";
+      upscale_models = "${modelsPath}/upscale_models";
+      vae = "${modelsPath}/vae";
+      vae_approx = "${modelsPath}/vae_approx";
+    };
+  };
+
+  modelPathsFile = writeTextFile {
+    name = "extra_model_paths.yaml";
+    text = (lib.generators.toYAML { } config-data);
+  };
+
+  pythonEnv = (
+    python312.withPackages (
```

Comment:
Do we really need to use 3.12 explicitly? Can't we use just 3?

Reply:
We need to use 3.12 explicitly for comfy